nifi-stateless/nifi-stateless-bundle/nifi-stateless-engine/src/main/java/org/apache/nifi/stateless/engine/StatelessFlowManager.java

```diff
@@ -320,9 +321,55 @@ public ReportingTaskNode createReportingTask(final String type, final String id,
 
     @Override
     public ParameterProviderNode createParameterProvider(final String type, final String id, final BundleCoordinate bundleCoordinate, final Set<URL> additionalUrls, final boolean firstTimeAdded,
-                                                         final boolean register) {
+                                                         final boolean registerLogObserver) {
+        if (type == null || id == null || bundleCoordinate == null) {
+            throw new NullPointerException("Must supply type, id, and bundle coordinate in order to create Reporting Task. Provided arguments were type=" + type + ", id=" + id
+                + ", bundle coordinate = " + bundleCoordinate);
+        }
+
+        // make sure the first reference to LogRepository happens outside of a NarCloseable so that we use the framework's ClassLoader
+        final LogRepository logRepository = LogRepositoryFactory.getRepository(id);
+
+        final ParameterProviderNode taskNode;
```

Comment:
The variable name should be changed to `providerNode` instead of `taskNode` to follow the naming convention of the class.
```suggestion
        final ParameterProviderNode providerNode;
```